### ISSUE TITLE
Read receipts always move the wrong receipt avatar on read in a group room

### ISSUE THREAD

**Issue by Half-Shot** [vector-im/riot-web#14600]
This has been a thing for ages, but it's finally annoyed me enough to create an issue (I did look around, couldn't find anything matching).

To reproduce:
- Open a room with at least 2 other people
- Talk in the room, ideally mentioning one of them.
- Watch as one of the avatars moves onto your message, to reflect that they have read it.
- **BUT WAIT**, that wasn't the person you mentioned. An eager reader...or maybe.
- Swap to another room and then back, and notice that the avatars are now correct.

So in summary, this is mildly annoying to the point where I no longer use the avatars as reference but assume that if *A* avatar moved, it was probably the person I actually talked to.

**Comment by turt2live** [comment]
Same cause as https://github.com/vector-im/riot-web/issues/9769
